src/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs

```diff
@@ -795,21 +796,36 @@ public IEnumerable<string> GetFiles(string path, string filter, bool recursive)
                         select p.Name;
                 });
         }
-
+        
         public IEnumerable<string> GetFullPaths(string fileName)
         {
             return ThreadHelper.JoinableTaskFactory.Run(async delegate
             {
                 await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                MicrosoftBuildEvaluationProject buildProject = EnvDTEProjectUtility.AsMicrosoftBuildEvaluationProject(EnvDTEProject);
-                return buildProject.Items.Where(
-                    projectItem =>
+                var paths = new List<string>();
+                var q = new Queue<EnvDTEProjectItems>();
+                q.Enqueue(EnvDTEProject.ProjectItems);
+                while (q.Count > 0)
+                {
+                    var items = q.Dequeue();
+                    foreach (EnvDTE.ProjectItem item in items)
                     {
-                        var itemFileName = Path.GetFileName(projectItem.EvaluatedInclude);
-                        return string.Equals(fileName, itemFileName, StringComparison.OrdinalIgnoreCase);
-                    })
-                    .Select(projectItem => Path.Combine(ProjectFullPath, projectItem.EvaluatedInclude));
+                        if (item.Kind == NuGetVSConstants.VsProjectItemKindPhysicalFile)
+                        {
+                            if (StringComparer.OrdinalIgnoreCase.Equals(item.Name, fileName))
+                            {
+                                paths.Add(item.FileNames[1]);
+                            }
+                        }
+                        else if (item.Kind == NuGetVSConstants.VsProjectItemKindPhysicalFolder)
+                        {
+                            q.Enqueue(item.ProjectItems);
+                        }
+                    }
+                }
+
+                return paths;
```

Comment:
The old code uses msbuild api, which is easier to work with. But unfortunately, not all DTE projects can be converted to msbuild project. So the new code uses DTE api.